papers/Collabra/paper.Rmd

```diff
@@ -0,0 +1,334 @@
+
+On the other hand, there are also outliers that cannot be detected by statistical tools, but should be found and removed. For example, if we are studying the effects of X on Y among teenagers and we have one observation from a 20-year-old, this observation might not be a _statistical outlier_, but it is an outlier in the _context_ of our research, and should be discarded to allow for valid inferences.
+
+## Winsorization
+
+_Removing_ outliers can in this case be a valid strategy, and ideally one would report results with and without outliers to see the extent of their impact on results. This approach however can reduce statistical power. Therefore, some propose a _recoding_ approach, namely, winsorization: bringing outliers back within acceptable limits [e.g., 3 MADs, @tukey1963less]. However, if possible, it is recommended to collect enough data so that even after removing outliers, there is still sufficient statistical power without having to resort to winsorization [@leys2019outliers].
+
+The _easystats_ ecosystem makes it easy to incorporate this step into your workflow through the `winsorize()` function of *{datawizard}*, a lightweight R package to facilitate data wrangling and statistical transformations [@patil2022datawizard]. This procedure will bring back univariate outliers within the limits of 'acceptable' values, based either on the percentile, the _z_ score, or its robust alternative based on the MAD.
+
+```{r winsorization}
+data[1501:1502, ]  # See outliers rows
+
+# Winsorizing using the MAD
+library(datawizard)
+winsorized_data <- winsorize(data, method = "zscore", robust = TRUE, threshold = 3)
+
+# Values > +/- MAD have been winsorized
+winsorized_data[1501:1502, ]
+```
+
+## The Importance of Transparency
+
+Once again, it is a critical part of a sound outlier treatment that regardless of which SOD method used, it should be reported in a reproducible manner. Ideally, the handling of outliers should be specified *a priori* with as much detail as possible, and preregistered, to limit researchers' degrees of freedom and therefore risks of false positives [@leys2019outliers]. This is especially true given that interesting outliers and random outliers are often times hard to distinguish in practice. Thus, researchers should always prioritize transparency and report all of the following information: (a) how many outliers were identified; (b) according to which method and criteria, (c) using which function of which R package (if applicable), and (d) how they were handled (excluded or winsorized, if the latter, using what threshold). If at all possible, (e) the corresponding code script along with the data should be shared on a public repository like the Open Science Framework (OSF), so that the exclusion criteria can be reproduced precisely.
+
+## What happens ater?
+
+See comment in the thread.
```

Comment:
It would be nice to have a discussion or some thoughts about what to do after outlier detection: re-run analyses? Show results with and without outliers? Describe outlying sample? I don't think there is one single best approach but we could at least mention some caveats and important things.

One thing we might highlight is the need for reporting the characteristics of the outliers (how many were removed, the percentage, and eventually the features). For instance, we could add an argument in `report_participants()`  and `report_sample()`, `outliers` that takes a vector of outliers / the output of check_outliers, and that would add a description of the outliers. 

In `report_participants`,  could  look like...  "Description of whole sample. Out of this sample, X (3.54%) of participants were flagged as outliers (demographic summary), leaving X participants in the final set (new demographics  without)."

in `report_sample()`: @strengejacke ideas how to present it?

Reply:
> It would be nice to have a discussion or some thoughts about what to do after outlier detection: re-run analyses? Show results with and without outliers?

We do have already a section called "[Handling Outliers](https://github.com/easystats/performance/blob/collabra_paper/papers/Collabra/paper.Rmd#L293)". Does that correspond to what you wanted?

> [...] But what should we do with these outliers once identified? [...]

And then we go about describing the different types of outliers, and what to do for each type of outlier, and whether to keep, exclude, or winsorize depending on the case.

> One thing we might highlight is the need for reporting the characteristics of the outliers (how many were removed, the percentage, and eventually the features)

We already kind of do that in the transparency section no? (I added a mention of the percentage though, thanks)

> Thus, researchers should always prioritize transparency and report all of the following information: (a) how many outliers were identified (including percentage); (b) according to which method and criteria, (c) using which function of which R package (if applicable), and (d) how they were handled (excluded or winsorized, if the latter, using what threshold).

> we could add an argument in report_participants() and report_sample(), outliers that takes a vector of outliers / the output of check_outliers, and that would add a description of the outliers.

We don't mention describing the characteristics of the outliers though, perhaps I could add a paragraph on that. But do we really need a new function, when we can subset on the data directly with the outlier object or as you say a vector of outliers directly, if you have it? 

``` r
report_participants(data[which(outliers),])
```

And I wonder if that assumes homogeneity of the outliers, which they should not be for random outliers. Even if they are, there would be no way of knowing unless you also report measures of homogeneity. But with so few observations as 1 to maybe 5 outliers, how meaningful would these data be? Especially if they are e.g., error outliers that are excluded, then any sample description would be meaningless. Maybe only for interesting outliers to try to figure out a pattern? So, please convince me why you think this is important :)